### PR TITLE
Upgrade bundled Gradle Enterprise Maven extension version.

### PR DESCRIPTION
**Pull request by c00ler**
Upgrade bundled Gradle Enterprise Maven extension version to `1.16`.

### CODE REVIEW

**Review by facewindu — COMMENTED** [review]
1.16.1 is coming in a few days.? Shouldn't we update to it instead?

File: README.adoc [hunk 1]
```diff
@@ -225,6 +225,7 @@ The following table shows the compatibility of the plugin version with Gradle En
 
 |===
 |Jenkins Gradle Plugin version  | Injected Gradle Enterprise Maven extension version | Injected Common Custom User Data Maven extension version  | Minimum supported Gradle Enterprise version
+|2.2-SNAPSHOT                   | 1.16                                               | 1.11.1                                                    | 2022.3
```
Comment (guylabs) [inline README.adoc:228]: I'd rather use something like:
```suggestion
|Next version                   | 1.16                                               | 1.11.1                                                    | 2022.3
```

**Review by guylabs — APPROVED** [review]
Similar to what Fracois said, let's use `1.16.1` once released and then do another release of the Jenkins plugin.

**Comment by c00ler** [comment]
@facewindu @guylabs Good point about the next version. I will keep the PR open until `1.16.1` is released and then just change `1.6 -> 1.6.1`.

**Review by sorin-florea — APPROVED** [review]

**Pull request merged**